src/edu/csus/ecs/pc2/list/ListUtilities.java

```diff
@@ -0,0 +1,167 @@
+// Copyright (C) 1989-2023 PC2 Development Team: John Clevenger, Douglas Lane, Samir Ashoo, and Troy Boudreau.
+package edu.csus.ecs.pc2.list;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+import edu.csus.ecs.pc2.core.LanguageUtilities;
+import edu.csus.ecs.pc2.core.model.IInternalContest;
+import edu.csus.ecs.pc2.core.model.Language;
+import edu.csus.ecs.pc2.core.model.Problem;
+import edu.csus.ecs.pc2.imports.ccs.IContestLoader;
+import edu.csus.ecs.pc2.ui.SubmissionSolutionList;
+import edu.csus.ecs.pc2.validator.clicsValidator.ClicsValidator;
+
+/**
+ * A set of utilities that operate on lists.
+ * 
+ * @author Douglas A. Lane <pc2@ecs.csus.edu>
+ *
+ */
+public class ListUtilities {
+
+	/**
+     * Returns list of files that match the SubmissionSolutionList (judging types location/list)
+     * 
+     * @param files list of files to be filtered
+     * @param submissionList a class that manages a number of SubmissionSampleLocation
```

Comment:
The parameter name specified in this `@param` statement still doesn't match the formal parameter name in the method...